File: apps/motoboy-pwa/src/lastSyncLabel.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { lastSyncLabel } from "./lastSyncLabel";

test("shows a first update message before data sync", () => {
  assert.equal(lastSyncLabel(null), "Aguardando primeira atualizacao.");
});

test("shows current update message for recent sync", () => {
  const now = new Date("2026-05-18T15:00:30.000Z");
  const syncedAt = new Date("2026-05-18T15:00:00.000Z");

  assert.equal(lastSyncLabel(syncedAt, now), "Atualizado agora.");
});

test("formats the last sync time in Sao Paulo timezone", () => {
  const now = new Date("2026-05-18T15:10:00.000Z");
  const syncedAt = new Date("2026-05-18T14:00:00.000Z");

  assert.equal(lastSyncLabel(syncedAt, now), "Ultima atualizacao as 11:00.");
});

File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/LastSyncLabelTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import java.time.Instant
import kotlin.test.Test
import kotlin.test.assertEquals

class LastSyncLabelTest {
  @Test
  fun `shows first update message before sync`() {
    assertEquals("Aguardando primeira atualizacao.", lastSyncLabel(null))
  }

  @Test
  fun `shows current update message for recent sync`() {
    val now = Instant.parse("2026-05-18T15:00:30Z")
    val syncedAt = Instant.parse("2026-05-18T15:00:00Z")

    assertEquals("Atualizado agora.", lastSyncLabel(syncedAt, now))
  }

  @Test
  fun `formats last sync in Sao Paulo timezone`() {
    val now = Instant.parse("2026-05-18T15:10:00Z")
    val syncedAt = Instant.parse("2026-05-18T14:00:00Z")

    assertEquals("Ultima atualizacao as 11:00.", lastSyncLabel(syncedAt, now))
  }
}
